src/modules/helpchan.ts

```diff
@@ -287,6 +285,61 @@ export class HelpChanModule extends Module {
 		);
 	}
 
+	@command({ inhibitors: [isTrustedMember] })
+	async claim(msg: Message, member: GuildMember) {
+		const helpUser = await HelpUser.findOne({
+			userId: member.id,
+		});
+		if (helpUser) {
+			return msg.channel.send(
+				`${member.displayName} already has an open help channel: <#${helpUser.channelId}>`,
+			);
+		}
+
+		const channelMessages = await msg.channel.messages.fetch({ limit: 50 });
+		const questionMessages = channelMessages.filter(
+			({ author, id }) => author.id === member.id && id !== msg.id,
+		);
+
+		const msgContent = questionMessages
+			.array()
+			.slice(0, 10)
+			.map(msg => msg.content)
+			.reverse()
+			.join('\n')
+			.slice(0, 2000);
+
+		const claimedChannel = msg.guild!.channels.cache.find(
+			channel =>
+				channel.type === 'text' &&
+				channel.parentID == categories.ask &&
+				channel.name.startsWith(this.CHANNEL_PREFIX) &&
+				!this.busyChannels.has(channel.id),
+		) as TextChannel | undefined;
```

Comment:
`as` is for asserting the type, use `claimedChannel: TextChan...` instead.

Reply:
The assertion is required. `find` doesn't narrow the result to `TextChannel`, so TS thinks it returns `GuildChannel | undefined`

Reply:
Oh I see now, discord.js has stuff to narrow with`#GuildChannel.type` but that doesn't work with `find`.